examples/structural_plasticity_with_stdp_2d.py

```diff
@@ -0,0 +1,267 @@
+                           label="expoisson")
+INoisePost = sim.Population(pop_size,
+                            sim.SpikeSourcePoisson,
+                            {'rate': e_rate, 'start': 0, 'duration': simtime},
+                            label="expoisson")
+
+# +---------------------------------------------------------------------------+
+# | Creation of connections                                                   |
+# +---------------------------------------------------------------------------+
+
+# Connection parameters
+JEE = 3.
+
+# Connection type between noise poisson generator and excitatory populations
+ee_connector = sim.OneToOneConnector()
+
+# Noise projections
+sim.Projection(
+    INoisePre, pre_pop, ee_connector, receptor_type='excitatory',
+    synapse_type=sim.StaticSynapse(weight=JEE * 0.05))
+sim.Projection(
+    INoisePost, post_pop, ee_connector, receptor_type='excitatory',
+    synapse_type=sim.StaticSynapse(weight=JEE * 0.05))
+
+# Additional Inputs projections
+for i in range(len(IAddPre)):
+    sim.Projection(
+        IAddPre[i], pre_pop, ee_connector, receptor_type='excitatory',
+        synapse_type=sim.StaticSynapse(weight=JEE * 0.05))
+for i in range(len(IAddPost)):
+    sim.Projection(
+        IAddPost[i], post_pop, ee_connector, receptor_type='excitatory',
+        synapse_type=sim.StaticSynapse(weight=JEE * 0.05))
+
+# Structurally plastic connection between pre_pop and post_pop
+stdp_model = sim.STDPMechanism(
+    timing_dependence=sim.SpikePairRule(
+        tau_plus=20., tau_minus=20.0, A_plus=0.02, A_minus=0.02),
+    weight_dependence=sim.AdditiveWeightDependence(w_min=0, w_max=1.))
+
+
+structure_model_with_stdp = sim.StructuralMechanism(
+    stdp_model=stdp_model,
+    weight=0,  # Use this weights when creating a new synapse
+    s_max=32,  # Maximum allowed fan-in per target-layer neuron
+    # grid=[pop_size, 1],
+    grid=[np.sqrt(pop_size), np.sqrt(pop_size)],  # spatial org of neurons
+    random_partner=True,  # Choose a partner neuron for formation at random,
+    # as opposed to selecting one of the last neurons to have spiked
+    f_rew=10 ** 4,  # Hz
+    sigma_form_forward=1.,  # spread of feed-forward connections
+    delay=10  # Use this delay when creating a new synapse
+)
+
+plastic_projection = sim.Projection(
+    pre_pop, post_pop,
+    sim.FixedProbabilityConnector(0.),  # No initial connections
+    synapse_type=structure_model_with_stdp,
+    label="structurally_plastic_projection"
+)
+
+# +---------------------------------------------------------------------------+
+# | Simulation and results                                                    |
+# +---------------------------------------------------------------------------+
+
+# Record neurons' potentials
+pre_pop.record(['v', 'spikes'])
+post_pop.record(['v', 'spikes'])
+
+# Run simulation
+sim.run(simtime)
+
+# Retrieve connectivity information from SpiNNaker
+
+pre_weights = []
```

Comment:
pycharm compaining that this could be rewritten as a list literal. think it means pre_weights = list()
might be worth doing.